core/src/main/java/org/apache/iceberg/CachingCatalog.java

```diff
@@ -105,14 +103,14 @@ public void delete(TableIdentifier tableIdentifier, Table table, RemovalCause ca
   }
 
   private Cache<TableIdentifier, Table> createTableCache(Ticker ticker) {
+    boolean hasExpiration = expirationIntervalMillis > 0;
     Caffeine<TableIdentifier, Table> cacheBuilder = Caffeine
         .newBuilder()
         .softValues()
-        .removalListener(identLoggingRemovalListener);
+        .removalListener(hasExpiration ? new MetadataTableInvalidatingRemovalListener() : identLoggingRemovalListener);
```

Comment:
Nit: all of the stuff that gets added due to expiration should ideally be inside of the `if (hasExpiration)` block or in the return statement after that section if it's for non-expiring cache (i.e. the `identLoggingRemovalListener` here).

Can you split this line into that block (and the block after it for the `identLoggingRemovalListener`.

Reply:
Sure! Just had to change to `Caffeine<Object, Object>` on cache creation as we don't have the type yet